grafana/dashboards/drive-stats.json

```diff
@@ -240,7 +240,7 @@
           "group": [],
           "metricColumn": "none",
           "rawQuery": true,
-          "rawSql": "WITH since as (\n\tSELECT date FROM positions\n\tWHERE car_id = $car_id\n\tORDER BY date ASC\n\tLIMIT 1\n),\nactual AS (\n\tSELECT\n\t\tdate_trunc('day', date)::date AS date,\n\t\tmax(odometer) - min(odometer) AS distance\n\tFROM positions\n\tWHERE car_id = $car_id\n\tGROUP BY 1\n),\nbase_line AS (\n\tSELECT date_trunc('day', dd)::date AS date\n  FROM generate_series((select date from since) , now(), '1 day'::interval) dd\n)\nSELECT \n  $__time(base_line.date), \n  convert_km(COALESCE(distance, 0)::numeric, '$length_unit') as \"distance_$length_unit\"\nFROM base_line\nLEFT JOIN actual ON actual.date = base_line.date\nWHERE $__timeFilter(base_line.date)\nORDER BY 1;",
+          "rawSql": "SELECT date_trunc('day', dd)::date AS time,\n       convert_km(drives.distance,'$length_unit') as distance\n  FROM generate_series ((SELECT min(end_date) FROM drives), now()::DATE, '1 day'::interval) dd\n  LEFT JOIN (\n        SELECT car_id ,\n               end_date::date AS date,\n               distance\n          FROM public.drives\n       ) drives\n    ON dd.date = drives.date\n  JOIN settings\n    ON drives.car_id = settings.id\nWHERE $__timeFilter(dd.date)\nAND car_id = $car_id\nORDER BY 1;",
```

Comment:
The reason why this query uses the positions is that trips can sometimes not be logged. The greater the selected period, the larger the potential discrepancy. In my case, it only differs by 100km, but that can be different depending on the user.

Perhaps instead of looking for min and max odometer values, the earliest and most recent positions (where odometer != NULL) could be used.

Reply:
So I have looked at ways to speed this up if we must use postiions... I have some ideas but the best I can do is save 20% off my run time. It works fine on my i5 server, but I can see Raspberry Pi's being slow. I will create a ticket about indexing then.... Let me rteview because I know some can be removed ex:

on positions
positions_drive_id_index (drive_id)
is covered by
positions_drive_id_elevation_idx (drive_id, elevation)

Reply:
@adriankumpf  question.... if i backup and restore it completely sets the indexing back to what it was before. If you modify indexing and someone runs the restore, will they lose the indexing that was added?